Content.Server/Radio/EntitySystems/HeadsetSystem.cs

```diff
@@ -86,21 +106,149 @@ private void OnExamined(EntityUid uid, HeadsetComponent component, ExaminedEvent
+    {
+        component.Channels.Clear();
+        foreach (EntityUid i in component.KeyContainer.ContainedEntities)
+        {
+            if (TryComp<EncryptionKeyComponent>(i, out var key))
+            {
+                UploadChannelsFromKey(component, key);
+            }
+        }
+    }
+
+    private void OnInteractUsing(EntityUid uid, HeadsetComponent component, InteractUsingEvent args)
+    {
+        if (!TryComp<ContainerManagerComponent>(uid, out var storage))
+            return;
+        if(!component.IsKeysUnlocked)
+        {
+            _popupSystem.PopupEntity(Loc.GetString("headset-encryption-keys-are-locked"), uid, args.User);
+            return;
+        }
+        if (TryComp<EncryptionKeyComponent>(args.Used, out var key))
+        {
+            if (component.KeySlots > component.KeyContainer.ContainedEntities.Count)
+            {
+                if (_container.TryRemoveFromContainer(args.Used) && InstallKey(component, args.Used, key))
+                {                    
+                    _popupSystem.PopupEntity(Loc.GetString("headset-encryption-key-successfully-installed"), uid, args.User);
+                    _audio.PlayPvs(component.KeyInsertionSound, args.Target);
+                }
+            }
+            else
+            {
+                _popupSystem.PopupEntity(Loc.GetString("headset-encryption-key-slots-already-full"), uid, args.User);
+            }
+        }
+        if (TryComp<ToolComponent>(args.Used, out var tool))
+        {
+            if (component.KeyContainer.ContainedEntities.Count > 0)
+            {
+                if (_toolSystem.UseTool(
+                    args.Used, args.User, uid,
+                    0f, 0f, new String[] { component.KeysExtractionMethod },
+                    doAfterCompleteEvent: null, toolComponent: tool)
+                )
+                {
+                    var contained = component.KeyContainer.ContainedEntities.ToArray<EntityUid>();
+                    foreach (var i in contained)
+                        component.KeyContainer.Remove(i);
+                    component.Channels.Clear();
+                    UpdateDefaultChannel(component);
+                    _popupSystem.PopupEntity(Loc.GetString("headset-encryption-keys-all-extracted"), uid, args.User);
+                    _audio.PlayPvs(component.KeyExtractionSound, args.Target);
+                }
+            }
+            else
+            {
+                _popupSystem.PopupEntity(Loc.GetString("headset-encryption-keys-no-keys"), uid, args.User);
+            }
+        }
+    }
+
+    private void UpdateDefaultChannel(HeadsetComponent component)
+    {
+        if (component.KeyContainer.ContainedEntities.Count >= 1)
+            component.DefaultChannel = EnsureComp<EncryptionKeyComponent>(component.KeyContainer.ContainedEntities[0])?.DefaultChannel;
+        else
+            component.DefaultChannel = null;
+    }
```

Comment:
DefaultChannel isn't really explained anywhere on why we need it or what it's really for.